engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/internal/BaseWindowedCharUpdateByOperator.java

```diff
@@ -0,0 +1,178 @@
+package io.deephaven.engine.table.impl.updateby.internal;
+
+import io.deephaven.api.updateby.OperationControl;
+import io.deephaven.chunk.Chunk;
+import io.deephaven.chunk.IntChunk;
+import io.deephaven.chunk.LongChunk;
+import io.deephaven.chunk.WritableCharChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.table.*;
+import io.deephaven.engine.table.impl.sources.*;
+import io.deephaven.engine.table.impl.updateby.UpdateByWindowedOperator;
+import io.deephaven.engine.table.impl.util.RowRedirection;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static io.deephaven.engine.rowset.RowSequence.NULL_ROW_KEY;
+import static io.deephaven.util.QueryConstants.*;
+
+public abstract class BaseWindowedCharUpdateByOperator extends UpdateByWindowedOperator {
+    protected final WritableColumnSource<Character> outputSource;
+    protected final WritableColumnSource<Character> maybeInnerSource;
+
+    // region extra-fields
+    // endregion extra-fields
+
+    protected abstract class Context extends UpdateByWindowedOperator.Context {
```

Comment:
There's a lot of commonality between this and the cumulative version.

Reply:
We looked. Cumulative `initializeUpdate` could move to the context. At that point, the only difference is the contexts, which could also have a common base class for common functionality. Base operators can converge and be generic on context type.